src/anim/binder/default-anim-binder.js

```diff
@@ -131,38 +131,40 @@ class DefaultAnimBinder {
 
     findNode(path) {
         if (this._mask) {
-            var shouldAnimateNode = false;
-            const entityPath = path.entityPath.join('/');
-            for (var i = 0; i < this._mask.length; i++) {
-                var maskItem = this._mask[i];
-                if (typeof maskItem === 'string') {
-                    if (entityPath.indexOf(maskItem) !== -1) {
-                        shouldAnimateNode = true;
-                    }
-                    break;
-                }
-                if (maskItem.children) {
-                    if (entityPath.indexOf(maskItem.path) !== -1) {
-                        shouldAnimateNode = true;
-                        break;
-                    }
-                } else if (maskItem.path === entityPath) {
+            let shouldAnimateNode = false;
```

Comment:
this whole code block inside `if (this._mask)` does not seem very efficient. It feels like some pre-computation should be done to make it a lot less intensive operation. Especially those joins and slices.